.github/workflows/benchmark-substrate.yml

```diff
@@ -0,0 +1,200 @@
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    # Docker Hub image that `container-job` executes in
+    container: node:18
+
+    # Service containers to run with `container-job`
+    services:
+      # Label used to access the service container
+      postgres:
+        # Docker Hub image
+        image: postgres:12-alpine
+        # Provide the password for postgres
+        env:
+          POSTGRES_PASSWORD: postgres
+        # Set health checks to wait until postgres has started
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+
+    steps:
+      #Check out
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 100
+          token: ${{ secrets.REPO_TOKEN }}
+
+      - name: Install PostgreSQL client
+        run: |
+          apt-get update
+          apt-get install --yes postgresql-client
+
+      # queries database with postgres client
+      - name: Setup historical database
+        run: psql -h postgres -d postgres -U postgres -c 'CREATE EXTENSION IF NOT EXISTS btree_gist;'
+        env:
+          PGPASSWORD: postgres
+
+      - name: Setup Node.js environment
+        uses: actions/setup-node@v2
+        with:
+          node-version: 18
+
+      - name: Install dependencies
+        run: npm i @subql/node@${{ github.event.inputs.indexerVersion }} -g
+
+
+      - name: Ensure reports path exists
+        run: |
+          output_dir="output/benchmark/"
+          if [ ! -d "$output_dir" ]; then
+            mkdir -p "$output_dir"
+            echo "Created directory: $output_dir"
+          else
+            echo "Benchmark report directory already exists: $output_dir"
+          fi
+
+      - name: Benchmarking
+        id: app
+        run: |
+          chmod +x .github/workflows/scripts/substrate.sh
+          .github/workflows/scripts/substrate.sh ${{github.event.inputs.benchmark-time}} ${{ github.event.inputs.deployment }} ${{ github.event.inputs.network-endpoint }} ${{ github.event.inputs.batch-size }} ${{ github.event.inputs.workers }} ${{ github.event.inputs.disableHistorical }} ${{ github.event.inputs.others }}
+        env:
+          # The hostname used to communicate with the PostgreSQL service container
+          DB_USER: postgres
+          DB_PASS: postgres
+          DB_DATABASE: postgres
+          DB_HOST: postgres
+          DB_PORT: 5432
+
+
+      - name: Extract actual sdk version
+        id: extract_version
+        run: |
+          version_line=$(grep -oE 'Current @subql/node version is.*' output/benchmark/indexing.log)
+          # Use cut to extract the version from the line
+          version=$(echo "$version_line" | grep -oE '[0-9]+\.[0-9]+\.[0-9]+')
+          echo "::set-output name=version::$version"    
+
+
+      - name: Query metadata get processed height
+        id: query_meta
+        env:
+          PGPASSWORD: postgres
+        run: |
```

Comment:
Is it worth moving these scripts to separate files? Then we at least get syntax hilighting